qcodes/tests/delegate/test_device.py

```diff
@@ -36,3 +39,27 @@ def test_device_meas(station, chip):
             datasaver.flush_data_to_database()
         assert len(
             datasaver.dataset.to_pandas_dataframe_dict()["device1_drain"]) == 10
+
+
+def test_device_with_channels(chip, station):
+    device = chip.channel_device
+
+    assert device.gate_1 == station.dac.ch01
+    assert device.gate_1.voltage.post_delay == 0.01
+    assert device.readout.source_parameters == (station.lockin.phase,)
+    assert device.readout() == 1e-5
```

Comment:
Would it make sense to extend the tests to also ensure that you can set with one parameter and get the sane result from the other one and the other way around

Reply:
Good point. Done!